proto/query.proto

```diff
@@ -287,6 +287,10 @@ message ExecuteOptions {
   }
 
   TransactionIsolation transaction_isolation = 9;
+
+  // no_cache_query_plan specifies if the query plan shoud be cached by vitess.
```

Comment:
Note that even if we switch the exposed `set` option, this needs to stay as-is so that the default value in the ExecuteOptions protobuf stays 0.